apps/easypid/src/crypto/salt.ts

```diff
@@ -0,0 +1,25 @@
+import { CredoWebCrypto, TypedArrayEncoder } from '@credo-ts/core'
+import type { EasyPIDAppAgent } from '@package/agent'
+
+const GENERIC_RECORD_SALT_ID = 'GENERIC_RECORD_SALT_ID'
+
+export const createSalt = async (agent: EasyPIDAppAgent) => {
+  const maybeSalt = await getSalt(agent)
+  if (maybeSalt) return maybeSalt
+
+  const crypto = new CredoWebCrypto(agent.context)
+
+  const saltBytes = crypto.getRandomValues(new Uint8Array(12))
+  const saltString = TypedArrayEncoder.toBase64URL(saltBytes)
+  await agent.genericRecords.save({ content: { salt: saltString }, id: GENERIC_RECORD_SALT_ID })
+  return saltString
+}
+
+const getSalt = async (agent: EasyPIDAppAgent): Promise<string | null> => {
```

Comment:
a salt should be unique per key. So I think we should provide a salt id. or we should name these methods specific to the context of the salt (so e.g. WalletServiceProviderPinKeySalt)

Reply:
`WalletServiceProviderPinKeySalt` thats fine. This will only be used for the device key, so renaming for the method would be good, yes!